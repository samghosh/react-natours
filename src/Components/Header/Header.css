.header {
  height: 95vh;
  background-image: url("../../Assets/hero.jpg");
  background-size: cover;
  background-position: top;
  position: relative;
  clip-path: polygon(0 0, 100% 0, 100% 75vh, 0 100%);
}
.header .logo-box {
  position: absolute;
  top: 40px;
  left: 40px;
}
.header .logo {
  height: 35px;
  animation: rotateLogo 1s ease-out 0.25s;
}
.header .logo:hover {
  animation: moveInRight 1s ease-out 0.25s;
}
.header .text-box {
  position: absolute;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
}
.header .text-box .heading-primary {
  color: #fff;
  text-transform: uppercase;
  backface-visibility: hidden;
  margin-bottom: 60px;
}
.header .text-box .heading-primary .heading-primary-main {
  display: block;
  font-size: 60px;
  font-weight: 400;
  letter-spacing: 35px;
  animation: moveInLeft 1s ease-out;
}
.header .text-box .heading-primary .heading-primary-sub {
  display: block;
  font-size: 20px;
  font-weight: 700;
  letter-spacing: 17.4px;
  animation: moveInRight 1s ease-out;
}
