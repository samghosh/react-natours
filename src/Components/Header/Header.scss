
.header { 
    height: 95vh;
    background-image: linear-gradient(
      to right bottom,
      rgba(126, 213, 111, 0.8), 
      rgba(40, 180, 133, 0.8)), 
      url('../../Assets/hero.jpg');
    background-size: cover;
    background-position: top;
    position: relative;
    clip-path: polygon(0 0, 100% 0, 100% 75vh, 0 100%);

    .logo-box {
      position: absolute;
      top: 40px;
      left: 40px;
    }

    .logo {
      height: 35px;

      &:hover {
        animation: moveInRight 1s ease-out;
      }
    }

    .text-box {
      position: absolute;
      top: 40%;
      left: 50%;
      transform: translate(-50%, -50%);
      text-align: center;

      .heading-primary {
        color: #fff;
        text-transform: uppercase;
        backface-visibility: hidden;
        margin-bottom: 60px;
    
        .heading-primary-main {
          display: block;
          font-size: 60px;
          font-weight: 400;
          letter-spacing: 35px;
          animation: moveInLeft 1s ease-out;
        }
  
        .heading-primary-sub {
          display: block;
          font-size: 20px;
          font-weight: 700;
          letter-spacing: 17.4px;
          animation: moveInRight 1s ease-out;
        }
      }
    }
  }

  @keyframes moveInLeft {
    0% {
      opacity: 0;
      transform: translateX(-100px);
    }

    80% {
      transform: translateX(10px);
    }

    100% {
      opacity: 1;
      transform: translate(0);
    }
  }

  @keyframes moveInRight {
    0% {
      opacity: 0;
      transform: translateX(100px);
    }

    80% {
      transform: translateX(-10px);
    }

    100% {
      opacity: 1;
      transform: translateX(0);
    }
  }