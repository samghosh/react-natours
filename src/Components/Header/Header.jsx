import React from 'react'; 
import { string } from 'prop-types';

// Importing the Compiled CSS of the Header Component
import './Header.css';

// Importing the Logo Icon from the Assets
import logo from '../../Assets/logo-white.png';

// Importing the Button Component
import Button from '../Button/Button';

<<<<<<< HEAD
const Header = ({headerLogo, HeaderPrimaryText, HeaderSubText, buttonText}) => {
=======
const Header = ({ headerLogo, HeaderPrimaryText, HeaderSubText, buttonText }) => {
>>>>>>> parent of 9432730... Project Updated to suit a different portfolio
    return (
        <>
            <header className="header">
                <div className="logo-box">
<<<<<<< HEAD
                    <img src={headerLogo} className="logo" alt="Logo" />
=======
                    <img src={ headerLogo } className="logo" alt="Logo" />
>>>>>>> parent of 9432730... Project Updated to suit a different portfolio
                </div>

                <div className="text-box">
                    <h1 className="heading-primary">
<<<<<<< HEAD
                        <span className="heading-primary-main">{HeaderPrimaryText}</span>
                        <span className="heading-primary-sub">{HeaderSubText}</span>
                    </h1>

                    <Button buttonText={buttonText}></Button>
=======
                        <span className="heading-primary-main" data-testid='header-primary-text'>{HeaderPrimaryText}</span>
                        <span className="heading-primary-sub" data-testid='header-sub-text'>{HeaderSubText}</span>
                    </h1>

                    <Button buttonText={ buttonText }></Button>
>>>>>>> parent of 9432730... Project Updated to suit a different portfolio
                </div>
            </header>
        </>
    )
};

Header.defaultProps = {
    headerLogo: logo
}

Header.propTypes = {
    buttonText: string.isRequired, 
    HeaderPrimaryText: string.isRequired, 
    HeaderSubText: string.isRequired,
    headerLogo: string
}

export default Header; 