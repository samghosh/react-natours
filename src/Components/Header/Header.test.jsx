import { render } from '@testing-library/react';
import React from 'react';
import Header from './Header';

it('matches snapshot', () => {
    const { asFragment } = render(
        <Header 
            HeaderPrimaryText='Header Primary Text' 
            HeaderSubText='Header Sub Text'
            buttonText='Header Button Text'
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it('should render the passed props properly', () => {
    const { getByTestId } = render(
        <Header 
            HeaderPrimaryText='Header Primary Text' 
            HeaderSubText='Header Sub Text'
            buttonText='Header Button Text'
        />
    );
    expect(getByTestId('header-primary-text').textContent).toBe('Header Primary Text');
    expect(getByTestId('header-sub-text').textContent).toBe('Header Sub Text');
    expect(getByTestId('button-text').textContent).toBe('Header Button Text');
});