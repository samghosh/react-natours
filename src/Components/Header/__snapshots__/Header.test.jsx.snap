// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="logo-box"
    >
      <img
        alt="Logo"
        class="logo"
        src="logo-white.png"
      />
    </div>
    <div
      class="text-box"
    >
      <h1
        class="heading-primary"
      >
        <span
          class="heading-primary-main"
          data-testid="header-primary-text"
        >
          Header Primary Text
        </span>
        <span
          class="heading-primary-sub"
          data-testid="header-sub-text"
        >
          Header Sub Text
        </span>
      </h1>
      <a
        class="btn btn-white"
        href="#"
      >
        <span
          data-testid="button-text"
        >
          Header Button Text
        </span>
      </a>
    </div>
  </header>
</DocumentFragment>
`;
